### DISCUSSION TITLE
Deploying Next with an API server

### DISCUSSION THREAD

**Discussion by AlexanderBachSchmidt** [vercel/next.js#13881] [Help]
Hi everyone!

I recently tried Next.js for the first time for a small project I’m doing. I really loved it and decided to try and move my other application to using it.

The application today is create-react-app on the front-end and express/Apollo-server/Passportjs/Postgres in the backend. They’re deployed together on a Heroku instance.

My goal was to achieve better performance, benefit from the automatic code splitting, deploy on Vercel and make it easier to develop in the future

I get a bit lost on how I should deploy the app for optimal performance? Deploy the front end on Next and keep Express on Heroku? Because after my understanding (and this is my first app), Express also servers static files.

What is the best way to handle a Next front end and Apollo/Passport/Postgres backend?

Thank you very much!

**Comment by thiagodebastos** [comment]
@AlexanderBachSchmidt hey mate did you figure this out? If so would you mind closing the issue and perhaps sharing what you found?

**Comment by AlexanderBachSchmidt** [comment]
Hey. No unfortunately I never figured it out and ended up with another solution. Thanks for following up!
